rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java

```diff
@@ -523,6 +523,11 @@ public String eth_getTransactionCount(String address, String blockId) {
 
     public Block getBlockByJSonHash(String blockHash) {
         byte[] bhash = stringHexToByteArray(blockHash);
+
+        if (bhash.length != Keccak256.HASH_LEN) {
+            throw invalidParamError("invalid argument 0: hex string has length " + (bhash.length * 2) + ", want " + (Keccak256.HASH_LEN * 2) + " for common.Hash");
```

Comment:
this one is gonna be returned to a user, isn't it?.. not sure user facing errors should be so detailed..

Reply:
This is how geth behaves, should we change it?